### PR TITLE
🧪 [testing] add missing tests for authMiddleware

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** The authMiddleware lacked a test file, leaving this critical component untested.
📊 **Coverage:** Covered edge cases such as missing `Authorization` header, invalid header format (missing "Bearer "), invalid token (rejected by the mocked `verify`), and valid token setting the context user. Mocking `verify` isolated the test logic.
✨ **Result:** Significant improvement in the reliability of token validation and context mapping. Tests now verify expected responses successfully for success and failure scenarios.

---
*PR created automatically by Jules for task [10429614907482943057](https://jules.google.com/task/10429614907482943057) started by @is0692vs*

### CODE REVIEW

File: apps/api/src/middleware/auth.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+});
+
+describe("authMiddleware", () => {
+    const SECRET = "test-secret";
+
+    const createApp = () => {
+        const app = new Hono<{ Bindings: Env, Variables: Variables }>();
+        app.use("*", async (c, next) => {
+            // Mock c.env for testing
+            c.env = { JWT_SECRET: SECRET } as Env;
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/middleware/auth.test.ts]: > medium
> 
> Casting a partial object with `as Env` makes the test brittle. If `authMiddleware` or another middleware in the chain is updated to use another property from `c.env`, the tests could fail with a runtime error that the type system wouldn't catch. To make the tests more robust and maintainable, it's better to provide a complete `Env` object, even with dummy values for properties not used in this specific test.
> 
> ```ts
>             c.env = {
>                 JWT_SECRET: SECRET,
>                 DB: undefined as any,
>                 BUCKET: undefined as any,
>                 GITHUB_CLIENT_ID: "",
>                 GITHUB_CLIENT_SECRET: "",
>                 FRONTEND_URL: "",
>             };
> ```

@Jules
